class/defaults.yml

```
@@ -1,3 +1,8 @@
 parameters:
   csi_driver_smb:
-    namespace: syn-csi-driver-smb
+    namespace: kube-system
+    charts:
+      csi-driver-smb: v0.6.0
```

It looks like current implementation is using arrays. But dicts with key/value are better mergeable through the hierarchy, whereas arrays are overwritten entirely.
At least you should provide the default empty array
```suggestion
      csi-driver-smb: v0.6.0
    pvs: []
```
Maybe consider also using the full name instead of an abbreviation as parameter name: `persistentVolumes`

---

> It looks like current implementation is using arrays. But dicts with key/value are better mergeable through the hierarchy, whereas arrays are overwritten entirely.

Not quite, array entries cannot be adjusted in the hierarchy, but new entries can be appended.

For this use case, I don't see a need to adjust existing array entries in the hierarchy, as all entries will most likely be given at the cluster level anyway.

I agree that the component should set the parameter to the empty array.